File: IrrgartenP4/irrgartenRuby/irrgarten/monster.rb
require_relative 'labyrinth_character'

module Irrgarten
  class Monster < LabyrinthCharacter
    INITIAL_HEALTH = 10

    def initialize(name, intelligence, strength)
      super(name, intelligence, strength, INITIAL_HEALTH)
    end

    def attack
      Dice.intensity(@strength)
    end

    def defend(received_attack)
      defensive_energy = Dice.intensity(@intelligence)
      if defensive_energy < received_attack
        got_wounded
        dead?
      else
        false
      end
    end

    def to_s
      "Monster[Name: #{@name}, Health: #{@health}, Pos: (#{@row}, #{@col})]"
    end
    
  end
end

File: IrrgartenP4/IrrgartenJava/src/irrgarten/Monster.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author moham
 */
// Monster.java

public class Monster extends LabyrinthCharacter {
    private static final float INITIAL_HEALTH = 5;

    public Monster(String name, float intelligence, float strength) {
        super(name, intelligence, strength, INITIAL_HEALTH);
    }

    @Override
    public float attack() {
        return Dice.intensity(strength);
    }

    @Override
    public boolean defend(float receivedAttack) {
        if (!dead()) {
            float defensiveEnergy = Dice.intensity(intelligence);
            if (defensiveEnergy < receivedAttack) {
                gotWounded();
            }
        }
        return dead();
    }
}
